ci/infra/testrunner/tests/test_node_reboot.py

```diff
@@ -4,58 +4,45 @@
 
 import pytest
 
-from tests.utils import wait
+from tests.utils import (check_pods_ready, node_is_ready, wait)
 
 from timeout_decorator import timeout
 
 logger = logging.getLogger("testrunner")
 
 
-def check_pods_ready(kubectl):
-    pods = json.loads(kubectl.run_kubectl('get pods --all-namespaces -o json'))['items']
-    for pod in pods:
-        pod_status = pod['status']['phase']
-        pod_name   = pod['metadata']['name']
-        assert pod_status in ['Running', 'Completed'], f'Pod {pod_name} status {pod_status} != Running or Completed'
+def check_node_is_ready(platform, kubectl, role, nr):
+    assert node_is_ready(platform, kubectl, role, nr)
 
 
 @pytest.mark.pr
-@pytest.mark.parametrize('node_type,node_number', [('master', 1), ('worker', 0)])
-def test_hard_reboot(deployment, platform, skuba, kubectl, node_type, node_number):
-    assert skuba.num_of_nodes(node_type) > node_number
+@pytest.mark.parametrize('role,node', [('master', 1), ('worker', 0)])
+def test_hard_reboot(deployment, platform, kubectl, role, node):
+    """ Reboots master and worker nodes and checks they are back ready.
+    For masters, don't master 1, as master 0 is expected to be the
```

Comment:
Looks like this sentence has some copy & paste error. Probably it was intented to be `For masters, reboot master 1,...` :wink:

Reply:
copy&paste considered harmful